Applications/Spire/Include/Spire/Ui/DropDownWindow.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef SPIRE_DROP_DOWN_WINDOW_HPP
+#define SPIRE_DROP_DOWN_WINDOW_HPP
+#include <QWidget>
+#include "Spire/Ui/DropShadow.hpp"
+
+namespace Spire {
+
+  //! Represents a pop-up window for displaying a widget.
+  class DropDownWindow : public QWidget {
+    public:
+
+      //! Constructs a DropDownWindow.
+      /*!
+        \param is_click_activated True if the DropDownWindow should be shown
+                                  or hidden in response to parent widget
+                                  mouse/keyboard activation events, false
+                                  otherwise.
+        \param parent The parent widget. Used to determine the position of the
+                      DropDownWindow.
+      */
+      explicit DropDownWindow(bool is_click_activated,
+        QWidget* parent = nullptr);
+
+      virtual ~DropDownWindow() = default;
```

Comment:
What happens if you remove this destructor?

Reply:
Aren't virtual destructors required when inheriting from another class to ensure that the base class is destructed properly?

Reply:
Good point, however, only the top-most class needs to define a virtual destructor. In this case the top-most class is ```QObject``` and it does define a virtual destructor. As such, all descendants of ```QObject``` will inherit that destructor and so you only need one if you intend to extend its behavior.